amd_openvx_extensions/amd_rpp/source/ResizeMirrorNormalizeTensor.cpp

```diff
@@ -0,0 +1,337 @@
+
+    data->srcDescPtr->dataType = RpptDataType::U8;
+    data->dstDescPtr->dataType = RpptDataType::U8;
+
+    // Set numDims, offset, n/c/h/w values for src/dst
+    data->srcDescPtr->numDims = 4;
+    data->dstDescPtr->numDims = 4;
+    data->srcDescPtr->offsetInBytes = 0;
+    data->dstDescPtr->offsetInBytes = 0;
+    data->srcDescPtr->n = data->nbatchSize;
+    data->srcDescPtr->h = data->maxSrcDimensions.height;
+    data->srcDescPtr->w = data->maxSrcDimensions.width;
+    data->srcDescPtr->c = ip_channel;
+    data->dstDescPtr->n = data->nbatchSize;
+    data->dstDescPtr->h = data->maxDstDimensions.height;
+    data->dstDescPtr->w = data->maxDstDimensions.width;
+    data->dstDescPtr->c = ip_channel;
+   
+	data->srcDescPtr->layout = RpptLayout::NHWC;
+	data->dstDescPtr->layout = RpptLayout::NHWC;
+
+	data->srcDescPtr->strides.nStride = ip_channel * data->srcDescPtr->w * data->srcDescPtr->h;
+	data->srcDescPtr->strides.hStride = ip_channel * data->srcDescPtr->w;
+	data->srcDescPtr->strides.wStride = ip_channel;
+	data->srcDescPtr->strides.cStride = 1;
+
+	data->dstDescPtr->strides.nStride = ip_channel * data->dstDescPtr->w * data->dstDescPtr->h;
+	data->dstDescPtr->strides.hStride = ip_channel * data->dstDescPtr->w;
+	data->dstDescPtr->strides.wStride = ip_channel;
+	data->dstDescPtr->strides.cStride = 1;
+
+
+    // Initialize ROI tensors for src/dst
+    data->roiTensorPtrSrc  = (RpptROI *) calloc(data->nbatchSize, sizeof(RpptROI));
+
+    // Set ROI tensors types for src/dst
+    data->roiType = RpptRoiType::XYWH;
+#if ENABLE_HIP
+    hipMalloc(&data->d_dstImgSize, data->nbatchSize * sizeof(RpptImagePatch));
+    hipMalloc(&data->d_roiTensorPtrSrc, data->nbatchSize * sizeof(RpptROI));
+#endif
+	refreshResizeMirrorNormalizeTensor(node, parameters, num, data);
+#if ENABLE_OPENCL
+	if(data->device_type == AGO_TARGET_AFFINITY_GPU)
+		rppCreateWithStreamAndBatchSize(&data->rppHandle, data->handle.cmdq, data->nbatchSize);
+#elif ENABLE_HIP
+	if (data->device_type == AGO_TARGET_AFFINITY_GPU)
+		rppCreateWithStreamAndBatchSize(&data->rppHandle, data->handle.hipstream, data->nbatchSize);
+#endif
+	if(data->device_type == AGO_TARGET_AFFINITY_CPU)
+		rppCreateWithBatchSize(&data->rppHandle, data->nbatchSize);
+
+	STATUS_ERROR_CHECK(vxSetNodeAttribute(node, VX_NODE_LOCAL_DATA_PTR, &data, sizeof(data)));
+	return VX_SUCCESS;
+}
+
+static vx_status VX_CALLBACK uninitializeResizeMirrorNormalizeTensor(vx_node node, const vx_reference *parameters, vx_uint32 num)
+{
+	ResizeMirrorNormalizeTensorLocalData * data;
+	STATUS_ERROR_CHECK(vxQueryNode(node, VX_NODE_LOCAL_DATA_PTR, &data, sizeof(data)));
+
+#if ENABLE_HIP
+    hipFree(data->d_dstImgSize);
+    hipFree(data->d_roiTensorPtrSrc);
+#endif
+#if ENABLE_OPENCL || ENABLE_HIP
+	if(data->device_type == AGO_TARGET_AFFINITY_GPU)
+		rppDestroyGPU(data->rppHandle);
+#endif
+	if(data->device_type == AGO_TARGET_AFFINITY_CPU)
+		rppDestroyHost(data->rppHandle);
+	free(data->mean);
```

Comment:
all these data should be passed from rocAL to amd_rpp using tensor(pinned memory) in updated tensor branch

Reply:
We will be passing this using tensors in the Tensor PR